molten-core/src/main/java/com/hotels/molten/core/mdc/MDCContextPropagatingSubscriber.java

```diff
@@ -30,12 +30,12 @@
  */
 @Slf4j
 final class MDCContextPropagatingSubscriber<T> extends AbstractNonFusingSubscription<T> {
-    private static final String MDC_CONTEXT_KEY = "X-MDC";
+    private static final String MDC_CONTEXT_KEY = "X-Molten-MDC";
     private final Context context;
 
     private MDCContextPropagatingSubscriber(CoreSubscriber<? super T> subscriber, Map<String, String> mdc) {
         super(subscriber);
-        this.context = subscriber.currentContext().put(MDC_CONTEXT_KEY, mdc);
+        this.context = mdc != null ? subscriber.currentContext().put(MDC_CONTEXT_KEY, mdc) : subscriber.currentContext();
```

Comment:
Mdc nullability is already checked at line 45 in the `decorate(...)` method.

Reply:
Removed check in fd3e6c3